### PR TITLE
create integration tests for risk factors

### PR DESCRIPTION
**Pull request by rmudambi**
### Description

- *Category*: test
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5078

### Changes and notes
Created some simple integration tests, that should be improved upon, but are better than nothing for now
Some minor changes to data_transformations and distributions to make the tests pass with scalar data

### Testing
All automated tests pass

### CODE REVIEW

**Review by albrja — APPROVED** [review]

File: src/vivarium_public_health/risks/data_transformations.py [hunk 1]
```diff
@@ -109,33 +112,15 @@ def get_exposure_data(
     else:
         value_columns = builder.data.value_columns()(f"{risk}.exposure")
 
-    if distribution_type == "dichotomous" and "cat2" in value_columns:
```
Comment (stevebachmeier) [inline src/vivarium_public_health/risks/data_transformations.py:112]: This just isn't a thing anymore?

File: src/vivarium_public_health/risks/data_transformations.py [hunk 2]
```diff
@@ -109,33 +112,15 @@ def get_exposure_data(
...
 
 
 def load_exposure_data(builder: Builder, risk: EntityString) -> pd.DataFrame:
     risk_component = builder.components.get_component(risk)
-    exposure_data = risk_component.get_data(
+    return risk_component.get_data(
         builder, builder.configuration[risk_component.name]["data_sources"]["exposure"]
     )
 
-    if isinstance(exposure_data, (int, float)):
```
Comment (stevebachmeier) [inline src/vivarium_public_health/risks/data_transformations.py:125]: same here - just not an issue?

File: src/vivarium_public_health/risks/data_transformations.py [hunk 3]
```diff
@@ -486,6 +471,9 @@ def validate_relative_risk_rebin_source(
 
 def validate_rebin_source(builder, risk: EntityString, data: pd.DataFrame):
 
+    if not isinstance(data, pd.DataFrame):
+        return
```
Comment (stevebachmeier) [inline src/vivarium_public_health/risks/data_transformations.py:475]: nit: maybe it's just a me thing but I dislike not saying the type of return (even if it's the implied None). We don't do it elsewhere so maybe worth doing just for consistency

File: tests/risks/test_base_risk.py [hunk 4]
```diff
@@ -78,26 +133,207 @@ def categorical_risk():
...
+    assert isinstance(distribution, EnsembleDistribution)
+
+    expected_distributions = set(distribution_weights.keys()) - {"glnorm"}
+    assert {"ensemble_distribution_weights"} == set(distribution.lookup_tables.keys())
+    assert expected_distributions == set(distribution.parameters.keys())
+
+    simulation.step()
+
+    # todo: use fuzzy checker to confirm that we are getting the expected results
+    print("We didn't runtime error - success!")
```
Comment (stevebachmeier) [inline tests/risks/test_base_risk.py:339]: lol

**Review by stevebachmeier — APPROVED** [review]

**Pull request merged**